### DISCUSSION TITLE
Is fastest node selection only on document loads and not query's?

### DISCUSSION THREAD

**Discussion by Gomez12** [ravendb/ravendb#16359] [Q&A] [answered]
If I look in the log mentioned in : https://github.com/ravendb/ravendb/discussions/15324 then it seems that the fastest node selection only looks at document loads and not at query's.

And in reality I seem to see the same principle, but it looks like it works out the wrong way ;)

We have a 3-node cluster and if one machine starts a heavy indexing task then it seems to take allmost all the traffic to itself while I would like it to be the least likely candidate.

If I look at the other issue, then I can understand that this behavior is happening, as when a server indexes a lot, there is a high change that that server also has many documents in memory, so there is a high possibility that it will be fastest to serve a document.
While then we send a query to it, the query is slow as it is probably going through the index which is being reindexed. At the very least, that server has a lot of i/o on it as it is indexing.

Is my assumption correct, and are you thinking about other ways to determine fastest node, or perhaps a policy lowest cpu load?

**Comment by garayx** [comment, marked as answer]
Hi

The fastest node selection is for read only operations (load or query),

The node is chosen by fastest response time from server, it doesn't consider machine resource usage.
